### PR TITLE
small refactors mostly to improve error handling and readability

### PR DESCRIPTION
**Pull request by finfinack**
Proposing a few changes to make the code slightly more readable (e.g. by returning earlier and removing indentation in GetSummary) and handle errors in a more go way where this wasn't already the case anyway.

### CODE REVIEW

**Comment by lazyfrosch** [comment]
The changes make sense for code style, may I asked the intentions of your work, as it is no functional change?

**Comment by finfinack** [comment]
Thanks! And of course, that's a fair question. I was looking around for hacktober contributions in projects which are primarily security related (as I come from that field), are in Go (because I like that language) and won't require me to spend hours getting familiar with a new codebase.

File: nagios/overall.go [hunk 1]
```diff
@@ -44,27 +44,25 @@ func (o *Overall) GetStatus() int {
...
-		if o.Summary == "" {
-			o.Summary = "No status information"
-		} else {
-			o.Summary = "states: " + strings.TrimSpace(o.Summary)
-		}
+		return "No status information"
 	}
-
-	return o.Summary
+	return "states: " + strings.TrimSpace(o.Summary)
```
Comment (lazyfrosch) [inline nagios/overall.go:65]: We should store it in o.Summary and return this.

```suggestion
	o.Summary = "states: " + strings.TrimSpace(o.Summary)
	return o.Summary
```

**Review by lazyfrosch — CHANGES_REQUESTED** [review]
I only noticed that one change required

**Review by lazyfrosch — APPROVED** [review]

File: nagios/overall.go (hunk 1 above)
Comment (finfinack) [reply nagios/overall.go]: Argh, sorry, of course! Thanks for catching this :-/

**Pull request merged**